README.md

```diff
@@ -1,18 +1,133 @@
-| :---------------------------------: | :----: | :-----: |
-| `terraform-azurerm-virtual-network` |        | `VNet`  |
\ No newline at end of file
+# Deploy HPCC Systems on Azure under Kubernetes
+
+NOTE: A tutorial of this Terraform for the developer, or others who are interested, can be found [here](documentation/hpcc-tf-for-developers.md).
+
+This is a slightly-opinionated Terraform module for deploying an HPCC Systems cluster on Azure's kubernetes service (aks).  The goal is to provide a simple method for deploying a cluster from scratch, with only the most important options to consider.
+
+The HPCC Systems cluster created by this module uses ephemeral storage, which is the default. This means the storage will be deleted when the cluster is deleted) But, you can also have Persistent Storage.  See the section titled [Persistent Storage](#persistent-storage), below.
+
+This repo is a fork of the excellent work performed by Godson Fortil.  The original can be found in branch, HPCC-27615 of [https://github.com/gfortil/Terraform-azurerm-hpcc].
+
+## Requirements
+
+* <font color="red">**Terraform**</font> This is a Terraform module, so you need to have Terraform installed on your system.  Instructions for downloading and installing Terraform can be found at [https://www.terraform.io/downloads.html](https://www.terraform.io/downloads.html).  Do make sure you install a 64-bit version of Terraform, as that is needed to accommodate some of the large random numbers used for IDs in the Terraform modules.
+
+* <font color="red">**helm**</font> Helm is used to deploy the HPCC Systems processes under Kubernetes.  Instructions for downloading and installing Helm are at [https://helm.sh/docs/intro/install](https://helm.sh/docs/intro/install/).
+
+* <font color="red">**kubectl**</font> The Kubernetes client (kubectl) is also required so you can inspect and manage the Azure Kubernetes cluster.  Instructions for download and installing that can be found at [https://kubernetes.io/releases/download/](https://kubernetes.io/releases/download/).  Make sure you have version 1.22.0 or later.
+
+* <font color="red">**Azure CLI**</font> To work with Azure, you will need to install the Azure Command Line tools.  Instructions can be found at [https://docs.microsoft.com/en-us/cli/azure/install-azure-cli](https://docs.microsoft.com/en-us/cli/azure/install-azure-cli).  Even if you think you won't be working with Azure, this module does leverage the command line tools to manipulate network security groups within kubernetes clusters.  TL;DR: Make sure you have the command line tools installed.
+
+* To successfully create everything you will need to have Azure's `Contributor` role plus access to `Microsoft.Authorization/*/Write` and `Microsoft.Authorization/*/Delete` permissions on your subscription.  You may have to create a custom role for this.  Of course, Azure's `Owner` role includes everything so if you're the subscription's owner then you're good to go.
```

Comment:
TLH: These requirements have been verified and are required.

Reply:
Can you validate this through testing, please?  Ideally, have an Azure user who does not have the additional permissions try to create a cluster using this TF code.

I am asking for additional testing because:  The additional permissions were an outcome of something I did when I was writing a version of this code.  That is not a requirement from any of the upstream code repositories, so it should not be a requirement in your derived code.

Reply:
I tried to deploy lite from an azure VM in a free trial personal azure account. I couldn't do it because a free trial azure account is not allowed to change quotas, and the terraform needs more vCPUs that the quota. 
Also, I mentioned that the terraform requires my azure VM to have EncryptionAtHost enabled, which I documented, under 'Requirements' in README.md. I'm going to try lite deployment in my virtualbox VM to see if the terraform requires something like EncryptionAtHost.

Reply:
How many vCPUs are available to a free trial account?  What size HPCC cluster did you try to create?  Is it possible to configure a tiny HPCC cluster that will work in a free trial account?

It seems to me that someone wanting to "kick the tires" will use a trial account, unless they are part of a business that already uses Azure, so finding a way to make this work in a constrained resource space seems important.

Reply:
Have attempted to valided using a free trial azure personal account. But, aks fails because I need higher quota of VMs for eastus. But, free trial accounts won't let you change quotas.

Reply:
I have temporarily removed the special role you had on your Azure user account that was required for my old TF modules.  Please test your TF code with your regular business account, which now has only a simple Contributor role, and see if the TF deploys correctly.  This will indicate whether any other user needs special permissions or not.

Reply:
Yes the special role is needed. Now, I get this error when I attempt to deploy aks:
Error: authorization.RoleAssignmentsClient#Create: Failure responding to request: StatusCode=403 -- Original Error: autorest/azure: Service returned an error. Status=403 Code="AuthorizationFailed" Message="The client 'HumphrTL@risk.regn.net' with object id '35cbdc79-7ef5-4d2c-9b59-61ec21d76aa9' does not have authorization to perform action 'Microsoft.Authorization/roleAssignments/write' over scope '/subscriptions/49219efc-701f-4c7e-a2ac-c600308a69e3/resourceGroups/app-tlhhpccplatform-sandbox-eastus2-19543/providers/Microsoft.Network/virtualNetworks/tlhhpcc-dev-eastus2-vnet/subnets/aks-hpcc-private/providers/Microsoft.Authorization/roleAssignments/280f0607-8f28-0a1c-63ea-a3735dfa0578' or the scope is invalid. If access was recently granted, please refresh your credentials."
 
   with module.aks.module.cluster.azurerm_role_assignment.network_contributor_network,
   on .terraform/modules/aks/modules/cluster/main.tf line 8, in resource "azurerm_role_assignment" "network_contributor_network":
    8: resource "azurerm_role_assignment" "network_contributor_network" {

Reply:
I have spoke with Godji and Milind. Godji said that he has the role of Owner and has never tried a role with fewer permissions than Owner. Milind says he has always used TFE to deploy the RBA hpcc terraform, but, believes if you use the terraform outside of TFE you would need elevated permissions.